Tests/MessagingInApp/State/InAppMessageStateTests.swift

```diff
@@ -19,21 +13,30 @@ class InAppMessageStateTests: IntegrationTest {
     override func setUp() {
         super.setUp()
         engineWebMock.underlyingView = UIView()
-        diGraphShared.override(value: engineWebProvider, forType: EngineWebProvider.self)
         MessagingInApp.shared.setEventListener(globalEventListener)
-        inAppMessageManager = InAppMessageManager(
+        inAppMessageManager = InAppMessageStoreManager(
             logger: diGraphShared.logger,
             threadUtil: diGraphShared.threadUtil,
             logManager: diGraphShared.logManager,
             gistDelegate: diGraphShared.gistDelegate
         )
+        diGraphShared.override(value: engineWebProvider, forType: EngineWebProvider.self)
+        diGraphShared.override(value: inAppMessageManager, forType: InAppMessageManager.self)
     }
 
     override func tearDown() {
         inAppMessageManager = nil
         super.tearDown()
     }
 
+    func dispatchAndWait(_ action: InAppMessageAction, timeout: TimeInterval = 10.0) async throws {
```

Comment:
Did this help in making CI consistent? If yes, then I'm good with the change. If not, we should first find the root cause and might consider reverting this if it didn't help?